src/main/java/io/fabric8/elasticsearch/plugin/acl/ACLDocumentManager.java

```diff
@@ -0,0 +1,287 @@
+
+    public ACLDocumentManager(final PluginClient client, final PluginSettings settings, final SearchGuardSyncStrategyFactory documentFactory) {
+        this.searchGuardIndex = settings.getSearchGuardIndex();
+        this.client = client;
+        this.documentFactory = documentFactory;
+    }
+    
+    @SuppressWarnings("rawtypes")
+    interface ACLDocumentOperation{
+        
+        void syncDocuments(Collection<SearchGuardACLDocument> docs);
+        
+        BulkRequest buildRequest(Client client, BulkRequestBuilder builder, Collection<SearchGuardACLDocument> docs) throws IOException;
+    }
+
+    @SuppressWarnings("rawtypes")
+    class SyncFromContextOperation implements ACLDocumentOperation {
+
+        private OpenshiftRequestContext context;
+
+        public SyncFromContextOperation(OpenshiftRequestContext context) {
+            this.context = context;
+        }
+        
+        @Override
+        public void syncDocuments(Collection<SearchGuardACLDocument> docs) {
+            LOGGER.debug("Syncing from context to ACL...");
+            for (SearchGuardACLDocument doc : docs) {
+                if(ConfigurationSettings.SEARCHGUARD_MAPPING_TYPE.equals(doc.getType())){
+                    RolesMappingSyncStrategy rolesMappingSync = documentFactory.createRolesMappingSyncStrategy((SearchGuardRolesMapping) doc);
+                    rolesMappingSync.syncFrom(context);
+                } else if(ConfigurationSettings.SEARCHGUARD_ROLE_TYPE.equals(doc.getType())) {
+                    RolesSyncStrategy rolesSync = documentFactory.createRolesSyncStrategy((SearchGuardRoles) doc);
+                    rolesSync.syncFrom(context);
+                }
+            }
+        }
+
+        @Override
+        public BulkRequest buildRequest(Client client, BulkRequestBuilder builder, Collection<SearchGuardACLDocument> docs) throws IOException{
+
+            for (SearchGuardACLDocument doc : docs) {
+                UpdateRequestBuilder update = client
+                        .prepareUpdate(searchGuardIndex, doc.getType(), SEARCHGUARD_CONFIG_ID)
+                        .setDoc(doc.toXContentBuilder());
+                if(doc.getVersion() != null) {
+                    update.setVersion(doc.getVersion());
+                }
+                builder.add(update.request());
+                if(LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("Built {} update request: {}", doc.getType(), XContentHelper.convertToJson(doc.toXContentBuilder().bytes(),true, XContentType.JSON));
+                }
+            }
+            return builder.request();
+        }
+        
+    }
+    
+    @SuppressWarnings("rawtypes")
+    class ExpireOperation implements ACLDocumentOperation {
+
+        private long now;
+
+        public ExpireOperation(long currentTimeMillis) {
+            this.now = currentTimeMillis;
+        }
+
+        @Override
+        public void syncDocuments(Collection<SearchGuardACLDocument> docs) {
```

Comment:
this method name is misleading - is it really `expireDocuments`

Reply:
It's an impl of an interface.  I will rename to make it more generic